Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+                      cons = [document nodesForXPath: @"//outlet" error: NULL];
+                      consen = [cons objectEnumerator];
+                      celem = nil;
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *attr = [celem attributeForName: @"destination"];
+                          NSString *val = [attr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [attr setStringValue: @"-1"];
+                            }
+                        }
+                    }
+                }
+
+              // Delete extra firstResponder...
+              /*
+              NSArray *ar = [document nodesForXPath: @"//customObject[@sceneMemberID=\"firstResponder\"]"
+                                              error: NULL];
+              if ([ar count] > 0)
+                {
+                  id node = [ar objectAtIndex: 0];
+                  [node detach];
+                }
+              */
+              
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
+                                     forKey: controllerId];
+                }
+              
+              RELEASE(document);
+            }
+        }
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+}
+
+- (NSMapTable *) _processSegues: (NSXMLDocument *)xmlIn
+{
+  NSMapTable *mapTable = [NSMapTable strongToWeakObjectsMapTable];
+  NSArray *connectionsArray = [xmlIn nodesForXPath: @"//connections"
+                                             error: NULL];
+  NSArray *array = [xmlIn nodesForXPath: @"//objects"
+                                  error: NULL];
+  NSXMLElement *objects = [array objectAtIndex: 0]; // get the "objects" section
+  NSArray *controllers = [objects nodesForXPath: @"windowController"
+                                          error: NULL];
+
+  NSString *uuidString = nil;
+  NSArray *docArray = [xmlIn nodesForXPath: @"document" error: NULL];
+  if ([docArray count] > 0)
+    {
+      NSXMLElement *docElem = (NSXMLElement *)[docArray objectAtIndex: 0];
+      NSXMLNode *a = [docElem attributeForName: @"uuid"];
+      NSString *value = [a stringValue];
+      if (value != nil)
+        {
+          return [_documentsMap objectForKey: value];
+        }
+      else
+        {
+          uuidString = [[NSUUID UUID] UUIDString];
+          NSXMLNode *new_uuid_attr = [NSXMLNode attributeWithName: @"uuid"
+                                                      stringValue: uuidString];
+          [docElem addAttribute: new_uuid_attr];
+        }
+    }
+  
+  NSString *src = nil;
```

Comment:
Declaring a variable in the middle of a method will create issues with old compilers or compilers in old mode. I am all for allowing this but we should make sure this is nonsense amongst GNUstep developers.

Reply:
Do we support any compilers which disallow this?